src/sidebar/components/group-list-item.js

```diff
@@ -22,53 +39,82 @@ function GroupListItem({ analytics, group }) {
     actions.focusGroup(group.id);
   };
 
-  const focusedGroupId = useStore(store => store.focusedGroupId());
-  const isSelected = group.id === focusedGroupId;
-  const groupOrgName = orgName(group);
+  const leaveGroup = () => {
+    const message = `Are you sure you want to leave the group "${group.name}"?`;
+    if (window.confirm(message)) {
+      analytics.track(analytics.events.GROUP_LEAVE);
+      groups.leave(group.id);
+    }
+  };
+
+  const toggleSubmenu = event => {
+    event.stopPropagation();
+
+    // Prevents group items opening a new window when clicked.
+    // TODO - Fix this more cleanly in `MenuItem`.
+    event.preventDefault();
+
+    setExpanded(!isExpanded);
+  };
+
+  // Close the submenu when any clicks happen which close the top-level menu.
+  const collapseSubmenu = () => setExpanded(false);
 
   return (
-    <div
-      className={classnames({
-        'group-list-item__item': true,
-        'is-selected': isSelected,
-      })}
-      onClick={focusGroup}
-      tabIndex="0"
-    >
-      {/* the group icon */}
-      <div className="group-list-item__icon-container">
-        {group.logo && (
-          <img
-            className="group-list-item__icon group-list-item__icon--organization"
-            alt={groupOrgName}
-            src={group.logo}
-          />
-        )}
-      </div>
-      {/* the group name */}
-      <div className="group-list-item__details">
-        <a
-          className="group-list-item__name-link"
-          href=""
-          title={
-            group.type === 'private'
-              ? `Show and create annotations in ${group.name}`
-              : 'Show public annotations'
-          }
-        >
-          {group.name}
-        </a>
-      </div>
-    </div>
+    <Fragment>
```

Comment:
Mentally parsing what kind of markup/DOM structure results from this rendering was a little challenging. It may help to leave a comment with the rough structure of what results here. I believe it may be:

```
<MenuItem />
{isExpanded && (<ul>...submenu</ul>
  {!isSelectable && (<p>URL restriction disclaimer</p>)}
}
```

Meaning that you end up with 1-3 sibling elements which consist of: `<MenuItem />` (always), `<ul>` (when expanded), `<p>` (if URL-restricted and expanded). Is that correct?

It can be derived from the JSX, of course, but for me the `<Fragment>`s make it harder to parse easily. Took me a few minutes and I had to run the branch and inspect the DOM to grok.